### PR TITLE
Add Stimulus controller for scroll events

### PR DESCRIPTION
**Pull request by aschempp**
I didn't recompile the JS files here since I still have Yarn 1 locally 😬

### CODE REVIEW

File: core-bundle/assets/controllers/scroll-controller.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+export default class extends Controller {
+    static classes = ['up', 'down']
+
+    initialize () {
+        this.lastScroll = 0;
+        this.onScroll = this.onScroll.bind(this)
+    }
+
+    connect () {
+        this.element.addEventListener('scroll', this.onScroll, { passive: true });
```
Comment (ausi) [inline core-bundle/assets/controllers/scroll-controller.js]: shouldn’t this be `window.addEventListener…`?

File: core-bundle/contao/templates/backend/be_main.html5 [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (ausi) [inline core-bundle/contao/templates/backend/be_main.html5]: unrelated change?

File: core-bundle/assets/controllers/scroll-controller.js (hunk 1 above)
Comment (aschempp) [reply core-bundle/assets/controllers/scroll-controller.js]: you're right, fixed in 8a0cb276b9c2b0aa232ffda20a34f5949dbc18be

File: core-bundle/contao/templates/backend/be_main.html5 (hunk 2 above)
Comment (aschempp) [reply core-bundle/contao/templates/backend/be_main.html5]: fixed in 8a0cb276b9c2b0aa232ffda20a34f5949dbc18be

**Review by ausi — APPROVED** [review]
I have no idea what `static classes = ['up', 'down']` does and how `data-contao--scroll-down-class` gets converted to `this.hasDownClass` and `this.downClass`. I guess that is stimulus magic?

The rest LGTM 🎉

**Comment by aschempp** [comment]
> I guess that is stimulus magic?


Correct 😎

https://stimulus.hotwired.dev/reference/css-classes

**Comment by leofeyer** [comment]
Thank you @aschempp.
